engine/client/secretprovider/env.go

```diff
@@ -0,0 +1,21 @@
+package secretprovider
+
+import (
+	"context"
+	"fmt"
+	"os"
+)
+
+func envProvider(_ context.Context, name string) ([]byte, error) {
+	v, ok := os.LookupEnv(name)
+	if !ok {
+		// Don't show the entire env var name, in case the user accidentally passed the value instead...
+		// This is important because users originally *did* have to pass the value, before we changed to
+		// passing by name instead.
+		if len(name) >= 4 {
+			name = name[:3] + "..."
+		}
+		return nil, fmt.Errorf("secret env var not found: %q", name)
+	}
+	return []byte(v), nil
+}
```

Comment:
Question :question: Should it be possible to get an optional secret? This is a problem today, but curious if you had any thoughts while working on the providers?

e.g. suppose I do `dagger call ... --my-secret=env:NON_EXISTENT` - when writing scripts (or using `dagger shell`), then it might be difficult to determine whether I should use this arg.

Not sure if it should be the default, but it would be nice if there was a way of skipping the non-existent secret somehow.